#! /bin/sh /usr/share/dpatch/dpatch-run
## 38_debuginfo.dpatch by Petter Reinholdtsen <pere@hungry.com>

Change debug output to print symbolic runlevel names as wellas a hex
value with the bit pattern representing the runlevels.

@DPATCH@
--- insserv-1.08.0.orig/listing.c
+++ insserv-1.08.0/listing.c
@@ -410,11 +410,11 @@
     list_for_each(tmp, d_start) {
 	dir_t * dir = getdir(tmp);
 	if (dir->script)
-	    fprintf(stderr, "%.2d %s 0x%.2x (%s)\n",
-		   dir->order, dir->script, dir->lvl, dir->name);
+	    fprintf(stderr, "%.2d %s 0x%.2x '%s' (%s)\n",
+		   dir->order, dir->script, dir->lvl, lvl2str(dir->lvl), dir->name);
 	else
-	    fprintf(stderr, "%.2d %s 0x%.2x (%%%s)\n",
-		   dir->order, dir->name, dir->lvl, *dir->name == '$' ? "system" : "guessed");
+	    fprintf(stderr, "%.2d %s 0x%.2x '%s' (%%%s)\n",
+		   dir->order, dir->name, dir->lvl, lvl2str(dir->lvl), *dir->name == '$' ? "system" : "guessed");
     }
 }
 #endif
